### ISSUE TITLE
'UNIQUE constraint failed' error during class assignment

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#4563]
### Observed behavior

Got a server error while trying to bulk-assign coaches to a class. The operation was not successful. However, after removing all users and re-adding them, it worked fine.

The issue does not seem to be easily reproduced.

### Expected behavior

Operation should not fail

### User-facing consequences

Users might not be able to complete the task

### Errors and logs

```
ERROR    Internal Server Error: /api/auth/role/
Traceback (most recent call last):
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: UNIQUE constraint failed: kolibriauth_role.user_id, kolibriauth_role.collection_id, kolibriauth_role.kind

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/viewsets.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/views.py", line 483, in dispatch
    response = self.handle_exception(exc)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/views.py", line 443, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/views.py", line 480, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/mixins.py", line 21, in create
    self.perform_create(serializer)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/mixins.py", line 26, in perform_create
    serializer.save()
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/serializers.py", line 728, in save
    self.instance = self.create(validated_data)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/serializers.py", line 700, in create
    self.child.create(attrs) for attrs in validated_data
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/serializers.py", line 700, in <listcomp>
    self.child.create(attrs) for attrs in validated_data
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/rest_framework/serializers.py", line 940, in create
    instance = ModelClass.objects.create(**validated_data)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/query.py", line 394, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/d/Projects/learning_equality/repos/k012x/kolibri/core/auth/models.py", line 184, in save
    super(AbstractFacilityDataModel, self).save(*args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/morango/models.py", line 438, in save
    super(SyncableModel, self).save(*args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/morango/utils/uuids.py", line 118, in save
    super(UUIDModelMixin, self).save(*args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/base.py", line 924, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/base.py", line 963, in _do_insert
    using=using, raw=raw)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/query.py", line 1076, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1112, in execute_sql
    cursor.execute(sql, params)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/Users/d/Projects/learning_equality/repos/k012x/.venv/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: UNIQUE constraint failed: kolibriauth_role.user_id, kolibriauth_role.collection_id, kolibriauth_role.kind
ERROR    "POST /api/auth/role/?1544140709442=1544140709442 HTTP/1.1" 500 9406
ERROR:django.server:"POST /api/auth/role/?1544140709442=1544140709442 HTTP/1.1" 500 9406
```

### Steps to reproduce

I have _not_ been able to reproduce.

That said, here are the steps I believe I took:

* create some users
* create a class and edit it
* click the 'enroll learners' button
* click the 'select all' checkbox at the top of the table and then 'confirm'
* click the 'assign coaches' button
* click the 'select all' checkbox at the top of the table and then 'confirm'

### Context

0.12

**Comment by jonboiser** [comment]
This may be a duplicate of #5321 (same stack trace as in that issue), which has been fixed in #5322.

**Comment by rtibbles** [comment]
Good catch, I think this is indeed the same issue.
